### PR TITLE
GSLUX-585: Layer comparator

### PR DESCRIPTION
**Pull request by AlitaBernachot**
### JIRA issue

https://jira.camptocamp.com/browse/GSLUX-585

### Description

- [x] Add slider bullet in layer manager (only for the first layer)
- [x] Add slider splitter on the map
- [x] Accessibility on splitter (handle tab/left/right/esc keydown)
- [x] Attach permalink params: _lc_ (for on/off) and _sliderRatio_
- [x] Check result in v3 integration

### CODE REVIEW

File: src/components/slider/slider-comparator.vue
Comment (mki-c2c) [inline src/components/slider/slider-comparator.vue:157]: removeEventListener

File: src/components/slider/slider-comparator.vue
Comment (mki-c2c) [inline src/components/slider/slider-comparator.vue:152]: why not remove event listeners on mouseup as done in old version ? What is the performance penalty ?

File: src/components/slider/slider-comparator.vue
Comment (mki-c2c) [inline src/components/slider/slider-comparator.vue:172]: if we have "space", there should be backspace (?) for opposite movement, but this is used for backwards navigation in browser history => maybe shift + space for left movement ??

File: src/components/slider/slider-comparator.vue
Comment (mki-c2c) [inline src/components/slider/slider-comparator.vue:175]: On testing the accessible commands, I would love the following behaviour: when selecting the split toggle button, move focus to slider handle, on pressing esc, move focus back to toggle button.

**Review by mki-c2c — APPROVED** [review]
Fine work.
The accessibility functions are particularly nice !

Overall, the slider movement as well as update of the permalink is a bit laggy, I am not quite sure why this is the case, maybe some debounce mechanism is needed before really updating all state related variables.
I think it is good enough for the moment and we should have a look at this laggy behaviour some time later.

**Review by mki-c2c — CHANGES_REQUESTED** [review]
Can we check together the unregistering of the listeners?

File: src/components/slider/slider-comparator.vue
Comment (AlitaBernachot) [reply src/components/slider/slider-comparator.vue]: Thx for noticing this.

File: src/components/slider/slider-comparator.vue
Comment (AlitaBernachot) [reply src/components/slider/slider-comparator.vue]: I have added a mouseUp handler to remove event listeners on mouse up. Can you test this improvement please? I feel it is slightly faster.

File: src/components/slider/slider-comparator.vue
Comment (AlitaBernachot) [reply src/components/slider/slider-comparator.vue:175]: Thx for the suggestion => added backspace handler.

File: src/components/slider/slider-comparator.vue
Comment (AlitaBernachot) [reply src/components/slider/slider-comparator.vue]: Added focus on slider hander for now. (to be improved later)

**Comment by AlitaBernachot** [comment]
@mki-c2c Thanks for your review and your suggestions.
Added some improvement regarding accessibility.
Created a dumb component (`splitterElement`) to seperate app logic and ui logic.

Can you check please?

File: src/components/layer-manager/layer-item/layer-item.spec.ts
Comment (tkohr) [inline src/components/layer-manager/layer-item/layer-item.spec.ts:25]: Really not important, just a spelling typo 'a' instead of 'o':
```suggestion
        isLayerComparatorOpen: false,
        displayLayerComparatorOpen: false,
```

**Review by tkohr — APPROVED** [review]
Nice, huge work that works really well on my side, thanks @AlitaBernachot!

Only managed to lose the synchronizing with the URL ratio param when moving very fast, but this does not seem blocking to me. Also noticed that the displayed part of the layer is on the left side of the slider (where the label is as well) on prod. I think we should align to this.

Don't forget to rebase before the lib build (I know we have to improve the workflow here).

**Comment by AlitaBernachot** [comment]
@mki-c2c can you review this PR and approve the changes (if ok with you) please?

**Review by mki-c2c — APPROVED** [review]
Same as tobias.
The slider has become decently smooth now. There is no more a frustrating behavior. Thanks.
And it's true, the labelling is weird, because the corresponding layer is on the opposite side (label left, layer right). As it has always been ??

**Comment by AlitaBernachot** [comment]
Thanks for your approval.

> Same as tobias. The slider has become decently smooth now. There is no more a frustrating behavior. Thanks. And it's true, the labelling is weird, because the corresponding layer is on the opposite side (label left, layer right). As it has always been ??

Label and layer are now on the same side, check my last commit: https://github.com/Geoportail-Luxembourg/luxembourg-geoportail/pull/55/commits/456bb783c4cf7f3b7e0f6c7a7da07b5bec0d3009

image

**Pull request merged**
